src/btree/bt_cursor.c

```diff
@@ -685,7 +691,7 @@ __wt_btcur_search_near(WT_CURSOR_BTREE *cbt, int *exactp)
          */
         if (leaf_found &&
           (cbt->compare == 0 || (cbt->slot != 0 && cbt->slot != cbt->ref->page->entries - 1)))
-            WT_ERR(__wt_cursor_valid(cbt, &upd, &valid));
+            WT_ERR(__wt_cursor_valid(cbt, cbt->tmp, WT_RECNO_OOB, &upd, &valid));
```

Comment:
Are we protected from that situation you mentioned before where the internal page gets evicted out from underneath us?

Reply:
Yes -- we're holding a hazard pointer now, in all cases. The problem before was that we were using `WT_CURSOR.key` from a page we'd read a long time ago, and we were no longer holding a hazard pointer on it.